src/main/java/org/testng/reporters/XMLUtils.java

```diff
@@ -112,7 +114,7 @@ public static void xmlOpen(IBuffer result, String indent, String tag,
 
   public static void xmlClose(IBuffer result, String indent, String tag, String comment) {
     result.append(indent).append("</").append(tag).append(">")
-        .append(comment != null ? comment : "")
+        .append(Strings.isNotNullAndNotEmpty(comment) ? comment : "")
```

Comment:
Maybe we can add a method like `String getValueOrEmpty(String)` which will return the value or empty if null

Reply:
Fixed.